File: app/src/main/java/com/abhi/lokaldemo/di/AppComponent.java
package com.abhi.lokaldemo.di;

import android.app.Application;

import com.abhi.lokaldemo.LokalApplication;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjection;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;
import dagger.android.support.DaggerApplication;

@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        AppModule.class,
        ActivityBuilder.class
})
public interface AppComponent extends AndroidInjector<DaggerApplication> {

    void inject(LokalApplication app);

    @Override
    void inject(DaggerApplication instance);

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder application(Application application);
        AppComponent build();
    }
}

File: app/src/main/java/com/abhi/lokaldemo/di/AppComponent.kt
package com.abhi.lokaldemo.di

import com.abhi.lokaldemo.LokalApplication
import dagger.Component
import dagger.android.AndroidInjectionModule
import dagger.android.AndroidInjector
import javax.inject.Singleton
import android.app.Application
import dagger.BindsInstance
import dagger.android.support.DaggerApplication


@Singleton
@Component(modules=[AndroidInjectionModule::class, AppModule::class, ActivityBuilder::class])
internal interface AppComponent : AndroidInjector<LokalApplication> {

    override fun inject(app: LokalApplication)

    fun inject(instance: DaggerApplication)

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent
    }

}